numpy/core/_dtype_ctypes.py

```diff
@@ -33,17 +33,43 @@ def _from_ctypes_array(t):
 
 
 def _from_ctypes_structure(t):
-    # TODO: gh-10533, gh-10532
-    fields = []
+    # TODO: gh-10533
     for item in t._fields_:
         if len(item) > 2:
             raise TypeError(
                 "ctypes bitfields have no dtype equivalent")
-        fname, ftyp = item
-        fields.append((fname, dtype_from_ctypes_type(ftyp)))
+    else:
+        if hasattr(t,"_pack_"):
+            formats_array = []
+            offsets_array = []
+            names_array = []
```

Comment:
I'd drop the `_array` part of the names here - it's confusing, because they're actually lists - and the `s` implies that it's a list anyway

Reply:
Done. Also added an additional test.